mm2src/coins/lightning/ln_events.rs

```diff
@@ -532,7 +532,13 @@ impl LightningEventHandler {
         let keys_manager = self.keys_manager.clone();
 
         let fut = async move {
-            let change_destination_script = Builder::build_p2witness(&my_address.hash).to_bytes().take().into();
+            let change_destination_script = match Builder::build_p2wpkh(my_address.hash()) {
+                Ok(script) => script.to_bytes().take().into(),
+                Err(_) => {
+                    error!("Error building change script");
```

Comment:
Please try to make errors as informative as possible, so it's better to include the error and the address in the message.

Reply:
fixed in [64dbf87](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1960/commits/64dbf873bbcc5a92d86e3bc107a9ad7d6db3b715)